pkg/core/xds/inspect/config.go

```diff
@@ -0,0 +1,138 @@
+package inspect
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	"github.com/josephburnett/jd/v2"
+
+	api_common "github.com/kumahq/kuma/api/openapi/types/common"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/test/xds"
+	util_proto "github.com/kumahq/kuma/pkg/util/proto"
+	"github.com/kumahq/kuma/pkg/xds/cache/cla"
+	xds_context "github.com/kumahq/kuma/pkg/xds/context"
+	"github.com/kumahq/kuma/pkg/xds/envoy"
+	"github.com/kumahq/kuma/pkg/xds/generator"
+	xds_hooks "github.com/kumahq/kuma/pkg/xds/hooks"
+	v3 "github.com/kumahq/kuma/pkg/xds/server/v3"
+	"github.com/kumahq/kuma/pkg/xds/sync"
+)
+
+type ProxyConfig map[string]interface{}
+
+type ProxyConfigInspector struct {
+	zone              string
+	meshContext       xds_context.MeshContext
+	snapshotGenerator *v3.TemplateSnapshotGenerator
+}
+
+func NewProxyConfigInspector(meshContext xds_context.MeshContext, zone string, hooks ...xds_hooks.ResourceSetHook) (*ProxyConfigInspector, error) {
+	return &ProxyConfigInspector{
+		zone:        zone,
+		meshContext: meshContext,
+		snapshotGenerator: &v3.TemplateSnapshotGenerator{
+			ResourceSetHooks:      hooks,
+			ProxyTemplateResolver: generator.DefaultTemplateResolver,
+		},
+	}, nil
+}
+
+func (p *ProxyConfigInspector) Get(ctx context.Context, name string, shadow bool) (ProxyConfig, error) {
+	proxyBuilder := &sync.DataplaneProxyBuilder{
+		Zone:          p.zone,
+		APIVersion:    envoy.APIV3,
+		IncludeShadow: shadow,
+	}
+
+	proxy, err := proxyBuilder.Build(ctx, model.ResourceKey{Name: name, Mesh: p.mesh()}, p.meshContext)
+	if err != nil {
+		return nil, err
+	}
+
+	envoyCtx := xds_context.Context{
+		Mesh: p.meshContext,
+		ControlPlane: &xds_context.ControlPlaneContext{
+			CLACache: &cla.Retriever{},
+			Secrets:  &xds.TestSecrets{},
```

Comment:
I think we shouldn't use in the code `TestSecret` as it is a test resource `"github.com/kumahq/kuma/pkg/test/xds"`

Reply:
Ah right, I'll create NoopSecrets